nuget/lib/dependabot/nuget/version.rb

```diff
@@ -74,8 +74,26 @@ def compare_prerelease_part(other)
 
         return -1 if prerelease_string && !other_prerelease_string
         return 1 if !prerelease_string && other_prerelease_string
+        return 0 if !prerelease_string && !other_prerelease_string
 
-        prerelease_string.<=>(other_prerelease_string)
+        split_prerelease_string = prerelease_string.split(".")
+        other_split_prerelease_string = other_prerelease_string.split(".")
+
+        split_prerelease_string.zip(other_split_prerelease_string).each do |lhs, rhs|
+          result = compare_dot_separated_part(lhs, rhs)
+          return result unless result.zero?
+        end
+
+        0
+      end
+
+      def compare_dot_separated_part(lhs, rhs)
+        return -1 if lhs.nil?
+        return 1 if rhs.nil?
+
+        return lhs.to_i <=> rhs.to_i if lhs.match?(/^\d+$/) && rhs.match?(/^\d+$/)
+
+        lhs.<=>(rhs)
```

Comment:
*Total nit that you're more than free to ignore!*

I know this is following the pattern present in this file, but it's a pretty unidiomatic way to write this, operators in ruby (`<`, `==` and also the `<=>` "spaceship" operator) are all actually "just" methods, which is why calling them like `foo.<=>(bar)` is possible. However, typically in ruby you wouldn't write code like `if 1.==(2)` even though it's syntactically valid, you would write `if 1 == 2`. Same goes here.

```suggestion
        lhs <=> rhs
```